src/components/@shared/FormInput/InputElement/ConsumerParameters/ConsumerParameterInput.tsx

```diff
@@ -37,10 +37,20 @@ export default function ConsumerParameterInput({
     )
   }
 
+  const getFieldValue = (fieldName: string, parameter: ConsumerParameter) => {
+    const valueKey = fieldName.split('.').slice(-1)[0]
+    const value = parameter[valueKey]
+
+    if (valueKey !== 'required' || typeof value === 'string') return value
+
+    return value ? 'required' : 'optional'
+  }
```

Comment:
suggestion(blocking): adding a quick comment why this is needed would make sense for people who are unfamiliar with the code.

also adding a comment on line 41 would imrpove readability

Reply:
Added a couple of comments, hopefully they are clear enough.